connector.rb

```diff
@@ -1,37 +1,2758 @@
+# rubocop:disable Metrics/BlockLength
+# rubocop:disable Style/IfUnlessModifier
+# rubocop:disable Style/PreferredHashMethods
+# rubocop:disable Style/SymbolProc
+# rubocop:disable Lint/UnusedBlockArgument
+
+# frozen_string_literal: true
+
 {
-  title: 'Template Partner Repo',
+  title: '4me',
 
   connection: {
+    fields: [
+      {
+        name: 'account',
+        label: '4me Account',
+        optional: false,
+        hint: 'The 4me account identifier'
+      },
+      {
+        name: 'instance',
+        label: '4me Instance',
+        hint: 'The 4me instance',
+        optional: false,
+        control_type: 'select',
+        options: [
+          ['Production', 'production'],
+          ['Quality Assurance', 'quality_assurance'],
+          ['Demo', 'demo'],
+          ['Custom domain', 'custom_domain']
+        ]
+      },
+      {
+        ngIf: 'input.instance == "production" || input.instance == "quality_assurance"',
+        name: 'region',
+        label: '4me region',
+        hint: 'The 4me region',
+        optional: false,
+        control_type: 'select',
+        options: [
+          ['Europe', 'default'],
+          ['Australia', 'au'],
+          ['United Kingdom', 'uk'],
+          ['United States', 'us'],
+          ['Switzerland', 'ch']
+        ]
+      },
+      {
+        ngIf: 'input.instance == "custom_domain"',
+        name: 'custom_domain_name',
+        label: 'Domain name',
+        hint: 'The 4me domain name',
+        optional: false,
+        control_type: 'text'
+      },
+      {
+        name: 'auth_method',
+        control_type: 'select',
+        hint: 'The 4me authentication method',
+        optional: false,
+        pick_list: [
+          ['Personal Access Token', 'bearer'],
+          ['OAuth2 (Client Credentials Grant)', 'oauth2_client_credentials']
+        ],
+        extends_schema: true
+      }
+    ],
+
     authorization: {
-      type: 'custom_auth'
+      type: 'multi',
```

Comment:
As we have limited support for `multi` auth in partner adapters we agreed to use previous approach. Can you, please, review and update PR accordingly if everything is correct?
```
connection: {
    fields: [
      {
        name: 'account',
        label: '4me Account',
        optional: false,
        hint: 'The 4me account identifier'
      },
      {
        name: 'instance',
        label: '4me Instance',
        hint: 'The 4me instance',
        optional: false,
        control_type: 'select',
        options: [
          ['Production', 'production'],
          ['Quality Assurance', 'quality_assurance'],
          ['Demo', 'demo'],
          ['Custom domain', 'custom_domain']
        ]
      },
      {
        ngIf: 'input.instance == "production" || input.instance == "quality_assurance"',
        name: 'region',
        label: '4me region',
        hint: 'The 4me region',
        optional: false,
        control_type: 'select',
        options: [
          ['Europe', 'default'],
          ['Australia', 'au'],
          ['United Kingdom', 'uk'],
          ['United States', 'us'],
          ['Switzerland', 'ch']
        ]
      },
      {
        ngIf: 'input.instance == "custom_domain"',
        name: 'custom_domain_name',
        label: 'Domain name',
        hint: 'The 4me domain name',
        optional: false,
        control_type: 'text'
      },
      {
        name: 'auth_method',
        control_type: 'select',
        hint: 'The 4me authentication method',
        optional: false,
        pick_list: [
          ['Personal Access Token', 'bearer'],
          ['OAuth2 (Client Credentials Grant)', 'oauth2_client_credentials']
        ],
      },
      {
        ngIf: 'input.auth_method == "bearer"',
        name: 'bearer_token',
        label: 'Personal Access Token',
        hint: 'The 4me bearer token',
        control_type: :password,
        optional: false,
      },
      {
        ngIf: 'input.auth_method == "oauth2_client_credentials"',
        name: 'client_id',
        label: 'Client ID',
        optional: false,
        hint: 'The 4me OAuth 2.0 client ID'
      },
      {
        ngIf: 'input.auth_method == "oauth2_client_credentials"',
        name: 'client_secret',
        label: 'Client secret',
        optional: false,
        control_type: :password,
        hint: 'The 4me OAuth 2.0 client secret'
      },
    ],

    authorization: {
      type: 'custom_auth',

      acquire: lambda do |connection|
        instance = connection['instance']
        region = connection['region']
        custom_domain_name = connection['custom_domain_name']

        token_url =
          case instance
          when 'production'
            case region
            when 'au'
              'https://oauth.au.4me.com/token'
            when 'uk'
              'https://oauth.uk.4me.com/token'
            when 'us'
              'https://oauth.us.4me.com/token'
            when 'ch'
              'https://oauth.ch.4me.com/token'
            else
              'https://oauth.4me.com/token'
            end
          when 'quality_assurance'
            case region
            when 'au'
              'https://oauth.au.4me.qa/token'
            when 'uk'
              'https://oauth.uk.4me.qa/token'
            when 'us'
              'https://oauth.us.4me.qa/token'
            when 'ch'
              'https://oauth.ch.4me.qa/token'
            else
              'https://oauth.4me.qa/token'
            end
          when 'demo'
            'https://oauth.4me-demo.com/token'
          else
            "https://oauth.#{custom_domain_name}/token"
          end

        request = post(token_url)
        request.headers('x-4me-Account': connection['account'])
        payload = {
          grant_type: 'client_credentials',
          client_id: connection['client_id'],
          client_secret: connection['client_secret']
        }
        request = request.payload(payload)
        response = request.request_format_www_form_urlencoded

        {
          access_token: response['access_token']
        }
      end,

      refresh_on: [401, 403],

      apply: lambda do |connection|
        auth_method = connection['auth_method']

        if current_url.include?('https://graphql.')
          case auth_method
          when 'bearer'
            bearer_token = connection['bearer_token']
            headers(Authorization: "Bearer #{bearer_token}") unless bearer_token.blank?
          when 'oauth2_client_credentials'
            access_token = connection['access_token']
            headers(Authorization: "Bearer #{access_token}") unless access_token.blank?
          end
        end
      end
    },

    base_uri: lambda do |connection|
      instance = connection['instance']
      region = connection['region']
      custom_domain_name = connection['custom_domain_name']

      case instance
      when 'production'
        case region
        when 'au'
          'https://graphql.au.4me.com'
        when 'uk'
          'https://graphql.uk.4me.com'
        when 'us'
          'https://graphql.us.4me.com'
        when 'ch'
          'https://graphql.ch.4me.com'
        else
          'https://graphql.4me.com'
        end
      when 'quality_assurance'
        case region
        when 'au'
          'https://graphql.au.4me.qa'
        when 'uk'
          'https://graphql.uk.4me.qa'
        when 'us'
          'https://graphql.us.4me.qa'
        when 'ch'
          'https://graphql.ch.4me.qa'
        else
          'https://graphql.4me.qa'
        end
      when 'demo'
        'https://graphql.4me-demo.com'
      else
        "https://graphql.#{custom_domain_name}"
      end
    end
  },
  ```

Reply:
@den-sergeyev-workato
I updated the `connection` to `custom_auth`,
recreated the VCR recordings. The `connector` `establishes invalid connection` spec failed on the old recordings,
and made a few small changes
- Few rubocop warnings
- Moved `auth_method = connection['auth_method']` into the if statement